tests/Proto.Actor.Tests/EventStreamTests.cs

```diff
@@ -48,5 +49,18 @@ public void EventStream_OnlyReceiveSubscribedToEventTypes()
             eventStream.Publish("not an int");
             Assert.Equal(0, eventsReceived.Count);
         }
+
+        [Fact]
+        public void EventStream_CanSubscribeToSpecificEventTypes_Async()
+        {
+            var received = "";
+            var eventStream = new EventStream();
+            eventStream.Subscribe<string>(theString =>
+            {
+                received = theString;
+                Assert.Equal("hello", received);
```

Comment:
public static ThreadPoolDispatcher DefaultDispatcher { get; } = new ThreadPoolDispatcher();

- uses ThreadPoolDispatcher by default, hence the test assertion doesn't get called, as the test completes before the assertion.